### PR TITLE
Make the Ioctl function common

### PR DESCRIPTION
**Pull request by nomis52**

### CODE REVIEW

File: firmware/src/rdm_responder.c [hunk 1]
```diff
@@ -104,13 +124,50 @@ static uint8_t *BuildSensorValueResponse(uint8_t *ptr, uint8_t index,
...
+  g_internal_state.mute_port = settings->mute_port;
+  g_internal_state.mute_bit = settings->mute_bit;
+
+  g_internal_state.identify_timer = 0u;
+  g_internal_state.identify_port = settings->identify_port;
+  g_internal_state.identify_bit = settings->identify_bit;
+
+  // Initialize hardware
+  PLIB_PORTS_PinDirectionOutputSet(
+    PORTS_ID_0, g_internal_state.identify_port, g_internal_state.identify_bit);
```
Comment (peternewman) [inline firmware/src/rdm_responder.c]: Indent by four from PLIB

File: firmware/src/rdm_responder.c [hunk 2]
```diff
@@ -104,13 +124,50 @@ static uint8_t *BuildSensorValueResponse(uint8_t *ptr, uint8_t index,
...
+  PLIB_PORTS_PinClear(PORTS_ID_0, g_internal_state.identify_port,
+                      g_internal_state.identify_bit);
+
+  PLIB_PORTS_PinDirectionOutputSet(
+    PORTS_ID_0, g_internal_state.mute_port, g_internal_state.mute_bit);
```
Comment (peternewman) [inline firmware/src/rdm_responder.c]: Again

File: firmware/src/rdm_responder.c [hunk 3]
```diff
@@ -284,6 +341,19 @@ int RDMResponder_DispatchPID(const RDMHeader *header,
   return RDMResponder_BuildNack(header, NR_UNKNOWN_PID);
 }
 
+int RDMResponder_Ioctl(ModelIoctl command, uint8_t *data, unsigned int length) {
```
Comment (peternewman) [inline firmware/src/rdm_responder.c]: Is this ever going to get some other commands? :)

File: firmware/src/rdm_responder.c [hunk 4]
```diff
@@ -728,8 +803,21 @@ int RDMResponder_GetIdentifyDevice(const RDMHeader *header,
 
 int RDMResponder_SetIdentifyDevice(const RDMHeader *header,
                                    const uint8_t *param_data) {
-  return RDMResponder_GenericSetBool(header, param_data,
+  bool previous_identify = g_responder->identify_on;
+  int r = RDMResponder_GenericSetBool(header, param_data,
                                      &g_responder->identify_on);
```
Comment (peternewman) [inline firmware/src/rdm_responder.c]: Indent one more.

File: firmware/src/simple_model.c [hunk 5]
```diff
@@ -203,7 +96,8 @@ static const PIDDescriptor PID_DESCRIPTORS[] = {
     RDMResponder_SetDeviceLabel},
   {PID_SOFTWARE_VERSION_LABEL, RDMResponder_GetSoftwareVersionLabel, 0u,
     (PIDCommandHandler) NULL},
-  {PID_IDENTIFY_DEVICE, RDMResponder_GetIdentifyDevice, 0u, SetIdentifyDevice}
+  {PID_IDENTIFY_DEVICE, RDMResponder_GetIdentifyDevice, 0u,
+    RDMResponder_SetIdentifyDevice}
```
Comment (peternewman) [inline firmware/src/simple_model.c]: I think these should all really be four.

**Comment by peternewman** [comment]
Fix Travis.

Generally LGTM

File: firmware/src/rdm_responder.c (hunk 1 above)
Comment (nomis52) [inline firmware/src/rdm_responder.c]: Done

File: firmware/src/rdm_responder.c (hunk 2 above)
Comment (nomis52) [inline firmware/src/rdm_responder.c]: Done

File: firmware/src/rdm_responder.c (hunk 3 above)
Comment (nomis52) [inline firmware/src/rdm_responder.c]: It'll get the DMX handler once I get back to adding proper SPI support

File: firmware/src/rdm_responder.c (hunk 4 above)
Comment (nomis52) [inline firmware/src/rdm_responder.c]: Done

File: firmware/src/rdm_responder.c (hunk 3 above)
Comment (peternewman) [inline firmware/src/rdm_responder.c]: Understood.

**Comment by peternewman** [comment]
LGTM

**Pull request merged**